Step04/app/src/main/java/com/kwoolytech/step04/MainActivity.java

```diff
@@ -34,34 +35,35 @@ protected void onCreate(Bundle savedInstanceState) {
         ActionBarDrawerToggle toggle = new ActionBarDrawerToggle(this, drawer, toolbar,
                 R.string.navigation_drawer_open, R.string.navigation_drawer_close);
 
-        drawer.setDrawerListener(toggle);
         toggle.syncState();
 
         NavigationView navigationView = (NavigationView)findViewById(R.id.nav_view);
         navigationView.setNavigationItemSelectedListener(this);
 
-        dataModel = new HansWeatherDataModel();
+        dataModel   = new HansWeatherDataModel();
         httpClient  = new KwoolyHttp(MainActivity.this);
 
         initializeUi();
         initializeKwoolyHttpCallbackFunction();
 
-        queryWeatherJsonData();
+        queryWeatherJsonData(37.49, 127.01);
 
```

Comment:
@kwoolytech how about using ```LocationManager``` to get the default location instead of using hardcoded value ?

If below API returns null, then you can fall back to any hardcoded value.
https://developer.android.com/reference/android/location/LocationManager.html#getLastKnownLocation(java.lang.String)